packages/compiler/src/server/diagnostics.ts

```diff
@@ -23,12 +28,32 @@ export function convertDiagnosticToLsp(
   program: Program,
   document: TextDocument,
   diagnostic: Diagnostic,
+  clientConfig?: Config | undefined,
 ): [VSDiagnostic, TextDocument][] {
-  const root = getVSLocation(
+  let root = getVSLocation(
     fileService,
     getSourceLocation(diagnostic.target, { locateId: true }),
     document,
   );
+
+  let message = diagnostic.message;
+  if (root === undefined) {
+    const emitterName = program.compilerOptions.emit?.find((emitName) =>
+      diagnostic.message.includes(emitName),
+    );
+    root = getLocationInTspConfig(fileService, program.compilerOptions.config, emitterName);
+    if (root === undefined) {
+      message +=
+        clientConfig?.lsp?.emit && emitterName && clientConfig.lsp.emit.includes(emitterName)
```

Comment:
it's better to have a comment here to provide some background about the possible error from ide setting

Reply:
and why is the fallback solution for linking other errors without location info at 0 position in corrent doc